test/app-luatest/console_debugger_session_test.lua

```diff
@@ -0,0 +1,128 @@
+local t = require('luatest')
+local popen = require('popen')
+local tnt = require('tarantool')
+local g = t.group()
+
+local function normalize_path(s)
+    return s:gsub("^@", ""):gsub("[^/]+$", "")
+end
+
+local function unescape(s)
+    return s:gsub('[\27\155][][()#;?%d]*[A-PRZcf-ntqry=><~]', '')
+end
+
+local function tarantool_path(arg)
+    local index = -2
+    -- arg[-1] is guaranteed to be non-null
+    while arg[index] do index = index - 1 end
+    return arg[index + 1]
+end
+
+local TARANTOOL_PATH = tarantool_path(arg)
+local path_to_script = normalize_path(debug.getinfo(1, 'S').source)
+local debug_target_script = path_to_script .. 'debug-target.lua'
+
+local DEBUGGER = 'luadebug.lua'
+local dbg_header = DEBUGGER .. ": Loaded for " .. tnt.version
+local dbg_prompt = DEBUGGER .. '>'
+
+local cmd_aliases = {
+    ['s'] = 's|st|step|step_into',
+    ['n'] = 'n|next|step_over',
+    ['f'] = 'f|finish|step_out',
+    ['p'] = 'p|print',
+    ['e'] = 'e|eval',
+    ['u'] = 'u|up',
+    ['d'] = 'd|down',
+    ['w'] = 'w|where',
+    ['t'] = 't|trace|bt',
+    ['l'] = 'l|locals',
+    ['h'] = 'h|help|?',
+    ['q'] = 'q|quit',
+}
```

Comment:
It was sorted in "natural" order :) (the order those commands were defined in `luadebug.lua` either in original 3rd party implementation, or in my refactored version. But I do agreed that sorted order is easier to navigate/lookup, thus I've sorted these commands both in `luadebug.lua`, in the test, and in the command builder.

```diff
diff --git a/test/app-luatest/console_debugger_session_test.lua b/test/app-luatest/console_debugger_session_test.lua
index 1af19631d..b85790807 100644
--- a/test/app-luatest/console_debugger_session_test.lua
+++ b/test/app-luatest/console_debugger_session_test.lua
@@ -32,18 +32,18 @@ local dbg_prompt = DEBUGGER .. '>'
 
 local cmd_aliases = {
     ['c'] = 'c|cont|continue',
-    ['s'] = 's|st|step|step_into',
-    ['n'] = 'n|next|step_over',
+    ['d'] = 'd|down',
+    ['e'] = 'e|eval',
     ['f'] = 'f|finish|step_out',
+    ['h'] = 'h|help|?',
+    ['l'] = 'l|locals',
+    ['n'] = 'n|next|step_over',
     ['p'] = 'p|print',
-    ['e'] = 'e|eval',
+    ['q'] = 'q|quit',
+    ['s'] = 's|st|step|step_into',
+    ['t'] = 't|trace|bt',
     ['u'] = 'u|up',
-    ['d'] = 'd|down',
     ['w'] = 'w|where',
-    ['t'] = 't|trace|bt',
-    ['l'] = 'l|locals',
-    ['h'] = 'h|help|?',
-    ['q'] = 'q|quit',
 }
 
 local MAX_ALIASES_COUNT = 4
diff --git a/third_party/lua/README-luadebug.md b/third_party/lua/README-luadebug.md
index e46265a18..ed6c7ef6d 100644
--- a/third_party/lua/README-luadebug.md
+++ b/third_party/lua/README-luadebug.md
@@ -57,19 +57,32 @@ Debugger Commands:
 
 If you have used other CLI debuggers, debugger.lua shouldn't be surprising. I didn't make a fancy parser, so the commands are just single letters. Since the debugger is pretty simple there are only a small handful of commands anwyay.
 
-	[return] - re-run last command
-	c(ontinue) - contiue execution
-	s(tep) - step forward by one line (into functions)
-	n(ext) - step forward by one line (skipping over functions)
-	p(rint) [expression] - execute the expression and print the result
-	f(inish) - step forward until exiting the current function
-	u(p) - move up the stack by one frame
-	d(own) - move down the stack by one frame
-	w(here) [line count] - print source code around the current line
-	t(race) - print the stack trace
-	l(ocals) - print the function arguments, locals and upvalues.
-	h(elp) - print this message
-	q(uit) - halt execution
+	c|cont|continue
+	- continue execution
+	d|down
+	- move down the stack by one frame
+	e|eval $expression
+	- execute the statement
+	f|finish|step_out
+	- step forward until exiting the current function
+	h|help|?
+	- print this help message
+	l|locals
+	- print the function arguments, locals and upvalues
+	n|next|step_over
+	- step forward by one line (skipping over functions)
+	p|print $expression
+	- execute the expression and print the result
+	q|quit
+	- exit debugger
+	s|st|step|step_into
+	- step forward by one line (into functions)
+	t|trace|bt
+	- print the stack trace
+	u|up
+	- move up the stack by one frame
+	w|where $linecount
+	- print source code around the current line
 
 If you've never used a command line debugger before, start a nice warm cozy fire, run tutorial.lua, and open it up in your favorite editor so you can follow along.
 
diff --git a/third_party/lua/luadebug.lua b/third_party/lua/luadebug.lua
index fedcd1bdd..869387519 100644
--- a/third_party/lua/luadebug.lua
+++ b/third_party/lua/luadebug.lua
@@ -530,20 +530,19 @@ end
 local gen_commands
 
 local function cmd_help()
-    for k, map in pairs(gen_commands) do
-        if map.first then
-            if #map.aliases > 0 then
-                local fun = require 'fun'
-                local txt = ''
-                fun.each(function(x) txt = txt .. '|' .. color_yellow(x) end,
-                            map.aliases)
-                print(color_blue(k) .. '|' .. string.sub(txt, 2, #txt) ..
-                        ' ' .. (map.arg or ''))
-            else
-                print(color_blue(k) .. ' ' .. (map.arg or ''));
-            end
-            print('    - ' .. map.help)
+    for _, v in ipairs(gen_commands) do
+        local map = gen_commands[v]
+        if #map.aliases > 0 then
+            local fun = require 'fun'
+            local txt = ''
+            fun.each(function(x) txt = txt .. '|' .. color_yellow(x) end,
+                     map.aliases)
+            print(color_blue(v) .. '|' .. string.sub(txt, 2, #txt) .. ' ' ..
+                  (map.arg or ''))
+        else
+            print(color_blue(v) .. ' ' .. (map.arg or ''));
         end
+        print('    - ' .. map.help)
     end
     return false
 end
@@ -555,18 +554,18 @@ end
 
 local commands_help = {
     {'c.ont.inue', 'continue execution', cmd_continue},
-    {'s.t.ep|step_into', 'step forward by one line (into functions)', cmd_step},
-    {'n.ext|step_over', 'step forward by one line (skipping over functions)',  cmd_next},
+    {'d.own', 'move down the stack by one frame',  cmd_down},
+    {'e.val $expression', 'execute the statement',  cmd_eval},
     {'f.inish|step_out', 'step forward until exiting the current function',  cmd_finish},
+    {'h.elp|?', 'print this help message',  cmd_help},
+    {'l.ocals', 'print the function arguments, locals and upvalues',  cmd_locals},
+    {'n.ext|step_over', 'step forward by one line (skipping over functions)',  cmd_next},
     {'p.rint $expression', 'execute the expression and print the result',  cmd_print},
-    {'e.val $expression', 'execute the statement',  cmd_eval},
+    {'q.uit', 'exit debugger', cmd_quit},
+    {'s.t.ep|step_into', 'step forward by one line (into functions)', cmd_step},
+    {'t.race|bt', 'print the stack trace',  cmd_trace},
     {'u.p', 'move up the stack by one frame',  cmd_up},
-    {'d.own', 'move down the stack by one frame',  cmd_down},
     {'w.here $linecount', 'print source code around the current line', cmd_where},
-    {'t.race|bt', 'print the stack trace',  cmd_trace},
-    {'l.ocals', 'print the function arguments, locals and upvalues',  cmd_locals},
-    {'h.elp|?', 'print this help message',  cmd_help},
-    {'q.uit', 'exit debugger', cmd_quit},
 }
 
 local function build_commands_map(commands)
@@ -602,7 +601,9 @@ local function build_commands_map(commands)
                     aliases = {},
                     arg = arg
                 }
-                if not first then
+                if first then
+                    table.insert(gen_commands, main_cmd)
+                else
                     assert(#main_cmd > 0)
                     table.insert(gen_commands[main_cmd].aliases, cmd)
                 end
```